### ISSUE TITLE
XmlSerialization of Angle and Length2d results in empty values

### ISSUE THREAD

**Issue by steingran** [angularsen/UnitsNet#569]
Scenario: When using XmlSerialization to serialize a class containing public properties of type Angle and/or Length2d, the values in these types are not preserved.

     public void Save([NotNull] Stream stream)
        {
            if (stream == null)
                throw new ArgumentNullException(nameof(stream));

            var xs = new XmlSerializer(GetType());
            xs.Serialize(stream, this);

            Saved?.Invoke();
        }

The method above in a class with Angle and/or Length2d as public properties results in empty values. Is this a supported scenario?

**Comment by steingran** [comment]
Never mind, I was made aware of https://github.com/angularsen/UnitsNet/issues/270. So no support for XML serialization, only JSON serialization

**Comment by angularsen** [comment]
Correct.

We chose to not support serialization for the same reasons as for BinarySerialization discussed here:
https://github.com/angularsen/UnitsNet/issues/564#issuecomment-441459146

TL;DR These types are not stable and should not be used for serialiation, we recommend creating your own types or use UnitsNet.Serialization.Json for JSON serialization. Happy to receive a PR though, if someone wants to take on supporting XML/Binary serialization too.
